domain/crossmodelrelation/state/model/remoteapplication.go

```diff
@@ -72,6 +74,59 @@ func (st *State) AddRemoteApplicationOfferer(
 	return nil
 }
 
+// AddRemoteApplicationConsumer adds a new synthetic application representing
+// the remote relation on the consuming model, to this, the offering model.
+func (st *State) AddRemoteApplicationConsumer(
+	ctx context.Context,
+	applicationName string,
+	args crossmodelrelation.AddRemoteApplicationConsumerArgs,
+) error {
+	db, err := st.DB(ctx)
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	if err := db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		// Check if the application already exists.
+		if err := st.checkApplicationNameAvailable(ctx, tx, applicationName); err != nil {
+			return errors.Errorf("checking if application %q exists: %w", applicationName, err)
+		}
+
+		// Check the offer doesn't already exist.
+		if err := st.checkOfferDoesNotExist(ctx, tx, args.OfferUUID); err != nil {
+			return errors.Capture(err)
+		}
+
+		// Create a remote relation for this consumer.
+		if err := st.insertRemoteRelation(ctx, tx, args.RelationUUID); err != nil {
+			return errors.Capture(err)
+		}
+
+		// Create an offer connection for this consumer.
+		offerConnectionUUID, err := st.insertOfferConnection(ctx, tx, args.OfferUUID, args.RelationUUID)
+		if err != nil {
+			return errors.Capture(err)
+		}
+
+		// Insert the application, along with the associated charm.
+		if err := st.insertApplication(ctx, tx, applicationName, args.AddRemoteApplicationArgs); err != nil {
```

Comment:
```suggestion
		synthApplicationName := fmt.Sprintf("remote-%s", relation.ApplicationToken)       
		if err := st.insertApplication(ctx, tx, applicationName, args.AddRemoteApplicationArgs); err != nil {
```

So we need to create a synthetic name here. We might want to do this in service layer, which I'm fine with, but just be aware.

https://github.com/juju/juju/blob/f303945110ddfef3a7cc8dbcffa2a9a81ee5d0b4/rpc/params/crossmodel.go#L503-L531

Reply:
Sure, this will be done in the service layer, on the next patch.